Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -122,31 +124,33 @@ void DVDInterface::DoState(PointerWrap& p)
   m_adpcm_decoder.DoState(p);
 }
 
-size_t DVDInterface::ProcessDTKSamples(std::vector<s16>* temp_pcm,
+size_t DVDInterface::ProcessDTKSamples(s16* target_samples, size_t block_count,
                                        const std::vector<u8>& audio_data)
 {
-  size_t samples_processed = 0;
-  size_t bytes_processed = 0;
-  while (samples_processed < temp_pcm->size() / 2 && bytes_processed < audio_data.size())
+  size_t available_block_count =
+      std::min(block_count, audio_data.size() / StreamADPCM::ONE_BLOCK_SIZE);
+  size_t blocks_processed = 0;
+  while (blocks_processed < available_block_count)
   {
-    m_adpcm_decoder.DecodeBlock(&(*temp_pcm)[samples_processed * 2], &audio_data[bytes_processed]);
+    const size_t sample_offset = blocks_processed * (StreamADPCM::SAMPLES_PER_BLOCK * 2);
```

Comment:
I have a weak preference for keeping these two offsets out of the loop as mutable variables and doing `+=` on each iteration. I think that makes it clearer how those values evolve over time ("they're indices that start at 0 and move by a given step amount") compared to the multiplication happening here.

Reply:
Fair enough. How about now?